src/enrichers/code_collection.py

```diff
@@ -12,16 +15,103 @@
+    GLOB = "glob"
+
+PATTERN_TYPE_FIELD_NAME = "patternType"
+PATTERN_FIELD_NAME = "pattern"
+IGNORE_CASE_FIELD_VALUE = "ignoreCase"
+MATCH_MODE_FIELD_NAME = "matchMode"
+ACTION_FIELD_NAME = "action"
+
+@dataclass
+class CodeBundleConfig:
+    pattern_type: PatternType
+    pattern_string: Optional[str]
+    ignore_case: bool
+    pattern: Optional[re.Pattern]
+    match_mode: StringMatchMode
+    action: CodeCollectionAction
+
+    @staticmethod
+    def construct_from_config(code_bundle_config: Union[str, dict[str, str]],
+                              defaults: "CodeBundleConfig" = None, pattern_required = True) -> "CodeBundleConfig":
+        if not defaults:
+            defaults = CODE_BUNDLE_CONFIG_DEFAULTS
+
+        if isinstance(code_bundle_config, str):
+            pattern_type = defaults.pattern_type
+            ignore_case = True
+            pattern_string = code_bundle_config
+            match_mode = defaults.match_mode
+            action = defaults.action
+            # If the config string starts with a !, then we reverse the sense of the action
+            # This is a convenience to make it easier to do a blacklist-style config that
+            # starts with a * inclusion and then a list of excluded bundle patterns without
+            # having to use the wordier dict-style config (which is parsed below in the
+            # else block).
+            if pattern_string[0] == '!':
+                pattern_string = pattern_string[1:]
+                action = CodeCollectionAction.EXCLUDE \
+                    if action == CodeCollectionAction.INCLUDE else CodeCollectionAction.INCLUDE
+        else:
+            pattern_type_string = code_bundle_config.get(PATTERN_TYPE_FIELD_NAME)
+            pattern_type = PatternType(pattern_type_string.lower()) if pattern_type_string else defaults.pattern_type
+            ignore_case = code_bundle_config.get(IGNORE_CASE_FIELD_VALUE, True)
+            pattern_string = code_bundle_config.get("pattern")
+            if pattern_required and pattern_string is None:
+                raise WorkspaceBuilderUserException('A code bundle config must include a "pattern" field to match')
+            match_mode_string = code_bundle_config.get(MATCH_MODE_FIELD_NAME)
+            match_mode = StringMatchMode(match_mode_string.lower()) if match_mode_string else defaults.match_mode
+            action_string = code_bundle_config.get(ACTION_FIELD_NAME)
+            action = CodeCollectionAction(action_string.lower()) if action_string else defaults.action
+
+        if pattern_type == PatternType.REGEX and pattern_string is not None:
+            pattern_flags = re.RegexFlag.IGNORECASE if ignore_case else 0
+            pattern = re.compile(pattern_string, pattern_flags)
+        else:
+            pattern = None
+
+        return CodeBundleConfig(pattern_type, pattern_string, ignore_case, pattern, match_mode, action)
+
+    def matches_code_bundle_name(self, code_bundle_name: str) -> bool:
```

Comment:
@vaterlaus I just did a quick test to validate the container would run and build docs, but got this right away: 

```
Discovering resources...
Error 500 from Workspace Builder service for command "run": 'str' object has no attribute 'matches_code_bundle_name'
```

Reply:
Resolved by latest commit